src/jit/codegencommon.cpp

```diff
@@ -4888,6 +4888,82 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #pragma warning(pop)
 #endif
 
+
+/*****************************************************************************
+* If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+* must be cleared to zeroes. The native compiler doesn't clear the upper bits
+* and there is no way to know if the caller is native or not. So, the upper
+* 32 bits of Vector argument on stack are always cleared to zero.
+*/
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+#if !defined(_TARGET_XARCH_)
+    return;
+#endif // !_TARGET_XARCH_
```

Comment:
To add further, JIT sticking to platform calling convention would mean any PInvoke/RPInvoke path only fixes (like the one i suggested earlier) won't be complete enough.  So clearing upper 4-bytes of the params in method prolog would be the way to go, which is what suggested by @CarolEidt and pursued by @sejongoh 

The only additional thing we need to handle is the case of PInvoke returning a Vector3 value.